### PR TITLE
Add to metadata files

### LINKED ISSUE #38: Add doi/id in file as a column
for attributes, etc.

### PR DESCRIPTION
**Pull request by drkrynstrng**
I added taxonomic rank values as well as the EML version number to the summary metadata file (completes #39). For simplicity I did not include the taxonomic rank names. For the EML version there are multiple ways to extract the number but I used the schemaLocation field. I think this would be the most robust option but correct me if I'm wrong. I also added the metadata PID to the attribute files (closes #38).

Note: I also changed the project settings to automatically remove trailing whitespace.

### CODE REVIEW

**Comment by drkrynstrng** [comment]
The eml2 package has been moved into the EML package. I will go ahead and change the dependency.

**Comment by drkrynstrng** [comment]
I noticed the EML version extraction does not seem to work well with data from the LTER network, but it works fine with ADC or KNB. Is there a better way to extract the EML version from the metadata?

File: R/download_d1_data.R [hunk 1]
```diff
@@ -79,54 +83,55 @@ download_d1_data <- function(data_url, path) {
...
+    attributeList <- suppressWarnings(EML::get_attributes(entity_data$attributeList, eml))
+
     meta_tabular <- tabularize_eml(eml) %>% tidyr::spread(name, value)
-    
+
     ## Summary metadata from EML (combine with general metadata later)
     entity_meta <- suppressWarnings(list(
       Metadata_ID = meta_id[[1]],
       Metadata_URL = metadata_nodes$data$url[1],
+      Metadata_EML_Version = paste(stringr::str_extract_all(meta_tabular$eml.version, "[0-9]", simplify = TRUE), collapse = "."),
```
Comment (brunj7) [inline R/download_d1_data.R]:
```suggestion
      Metadata_EML_Version = str_extract_all(meta_tabular$eml.version, "[0-9].+[\\s]", simplify = TRUE),
```

**Comment by brunj7** [comment]
This PR should also solve #98

File: R/download_d1_data.R (hunk 1 above)
Comment (drkrynstrng) [reply R/download_d1_data.R]: I went through the regex more carefully and found a simple solution that works across nodes:
```
Metadata_EML_Version = stringr::str_extract(meta_tabular$eml.version, "\\d\\.\\d\\.\\d")
```

**Pull request merged**